pkg/controllers/clusterresourceplacementeviction/controller.go

```diff
@@ -0,0 +1,300 @@
+	for i := range crbList.Items {
+		if crbList.Items[i].Spec.TargetCluster == eviction.Spec.ClusterName {
+			if evictionTargetBinding == nil {
+				evictionTargetBinding = &crbList.Items[i]
+			} else {
+				klog.V(2).InfoS("Found more than one cluster resource binding for cluster targeted by eviction",
+					"clusterResourcePlacementEviction", evictionName, "clusterResourcePlacement", eviction.Spec.PlacementName)
+				markEvictionInvalid(&eviction, evictionInvalidMultipleCRBMessage)
+				return runtime.Result{}, r.updateEvictionStatus(ctx, &eviction)
+			}
+		}
+	}
+	if evictionTargetBinding == nil {
+		klog.V(2).InfoS("Failed to find cluster resource binding for cluster targeted by eviction",
+			"clusterResourcePlacementEviction", evictionName, "targetCluster", eviction.Spec.ClusterName)
+		markEvictionInvalid(&eviction, evictionInvalidMissingCRBMessage)
+		return runtime.Result{}, r.updateEvictionStatus(ctx, &eviction)
+	}
+
+	markEvictionValid(&eviction)
+
+	// Check to see if binding is being deleted.
+	if evictionTargetBinding.GetDeletionTimestamp() != nil {
+		klog.V(2).InfoS("ClusterResourceBinding targeted by eviction is being deleted",
+			"clusterResourcePlacementEviction", evictionName, "clusterResourceBinding", evictionTargetBinding.Name, "targetCluster", eviction.Spec.ClusterName)
+		markEvictionExecuted(&eviction, evictionAllowedPlacementRemovedMessage)
+		return runtime.Result{}, r.updateEvictionStatus(ctx, &eviction)
+	}
+
+	// Check to see if binding has failed. If so no need to check disruption budget we can evict.
+	if bindingutils.HasBindingFailed(evictionTargetBinding) {
+		klog.V(2).InfoS("ClusterResourceBinding targeted by eviction is in failed state",
+			"clusterResourcePlacementEviction", evictionName, "clusterResourceBinding", evictionTargetBinding.Name, "targetCluster", eviction.Spec.ClusterName)
+		if err := r.deleteClusterResourceBinding(ctx, evictionTargetBinding); err != nil {
+			return runtime.Result{}, err
+		}
+		markEvictionExecuted(&eviction, evictionAllowedPlacementFailedMessage)
+		return runtime.Result{}, r.updateEvictionStatus(ctx, &eviction)
+	}
+
+	isDBPresent := true
+	var db placementv1alpha1.ClusterResourcePlacementDisruptionBudget
+	if err := r.Client.Get(ctx, types.NamespacedName{Name: crp.Name}, &db); err != nil {
+		if !errors.IsNotFound(err) {
+			return runtime.Result{}, controller.NewAPIServerError(true, err)
+		}
+		isDBPresent = false
+	}
+
+	if !isDBPresent {
+		if err := r.deleteClusterResourceBinding(ctx, evictionTargetBinding); err != nil {
+			return runtime.Result{}, err
+		}
+		markEvictionExecuted(&eviction, evictionAllowedNoPDBMessage)
+		return runtime.Result{}, r.updateEvictionStatus(ctx, &eviction)
+	}
+
+	bindingList := crbList.Items
+	var desiredBindings int
+	switch crp.Spec.Policy.PlacementType {
+	case placementv1beta1.PickAllPlacementType:
+		desiredBindings = calculateSchedulerTargetedBindings(bindingList)
```

Comment:
Pick all is different (https://kubernetes.io/docs/tasks/run-application/configure-pdb/#arbitrary-controllers-and-selectors)